File: js/getHashTags.mjs
import axios from 'axios';
import cheerio from 'cheerio';

function parseHTML(html){
  const $ = cheerio.load(html);  
  const hashtags = $('div.tag-box').text();

  let matches = hashtags.match(/#\w+/g);
  if (matches) {
      matches = matches.slice(0, 30);
  }
  return matches;
}

/**
 * @param {string} query 
 * @returns array of hashtags, -1 if error
 */
export async function getHash(query){
  query = query.replace(/\s/g, '');
  const options = {
    method: 'GET',
    url: `https://best-hashtags.com/hashtag/${query}/`,
  };

  try {
    const response = await axios.request(options);
    return(parseHTML(response.data));
  } catch (error) {
    console.error(error.stack);
    return -1;
    process.exit();
  }
}

let query;
process.argv[2] !== undefined ? query = process.argv[2] : process.exit(1);

let res;
try{
  res = await getHash(query);
} catch (error) {
  console.error(error.stack);
  process.exit();
}

console.log(res.join(" "))

File: src/get_hashtags.rs
use scraper::{Selector, Html};
use regex::Regex;

fn parse_html(html: &str) -> Vec<String> {
    let fragment = Html::parse_fragment(html);
    let selector = Selector::parse("div.tag-box").unwrap();
    let mut hashtags = String::new();

    for element in fragment.select(&selector) {
        hashtags.push_str(&element.text().collect::<String>());
    }

    let re = Regex::new(r"#\w+").unwrap();
    let mut matches: Vec<String> = re.find_iter(&hashtags)
        .map(|mat| mat.as_str().to_string())
        .collect();

    matches.truncate(30);
    matches
}

pub fn ask_website (topic: String){
    let client = reqwest::Client::new();
    let url = "https://best-hashtags.com/hashtags/".to_string() + &topic;

    let response = client.post(url).send().await?.text().await?;
    println!("{}", response);
}